README.md

```diff
@@ -1251,6 +1252,26 @@ Note that Terragrunt will respect any `TF_VAR_xxx` variables you've manually set
 anything in `inputs` will NOT be override anything you've already set in your environment.  
 
 
+### Locals
+
+You can use locals to bind a name to values and expressions that you would like to create a reference for within your
+config. For example, suppose that you need to use the AWS region in multiple inputs. You can bind the name `aws_region`
+using locals:
+
+```
+locals = {
+  aws_region = "us-east-1"
+}
+
+inputs = {
+  aws_region  = local.aws_region
+  s3_endpoint = "com.amazonaws.${local.aws_region}.s3"
+}
+```
+
+You can use any valid terragrunt expression in the `locals` configuration.
```

Comment:
Prepare yourself for the "how do I reuse locals from another file?" question in 3... 2... 1... :-D

Perhaps tackle that head on and add a section on reusing/merging `yaml` files for now? And when we add `get_input` and `get_output`, we can replace it.

Reply:
Ok I can add that in.

FWIW, I was thinking about proposing a similar construct, but called `globals` which can be merged with the parent in an `include` call. I think the explicit separating of `locals` from `globals` is important, especially if and when `include` becomes more complex to allow inclusion of multiple levels of terragrunt config.

Reply:
Do you have any immediate plans to implement `globals`?  We've just developed a need for something like that (having variables in our parent `terragrunt.hcl` that we can override from child `terragrunt.hcl`s).  While including a yaml file if it exists would work, it doesn't seem like the cleanest solution and we'd have to walk it back once there's a better way.

If you haven't started and you think it would be something that would be accepted, I don't mind creating a PR for it.

Reply:
> If you haven't started and you think it would be something that would be accepted, I don't mind creating a PR for it.

I don't think we have immediate plans for `globals`. I would suggest first creating an issue (with `Feature Request` in the title) describing:

- Your use case. Why do you need something like `globals`?
- Ideal solution. Is `globals` what you want, or do you want something like `get_local`, which can take another `terragrunt.hcl` config file as input, a `key` and return the value assigned to that `local` in the target file?
- Indicate whether you want to open a PR, either for the ideal solution or if that is challenging, a temporary workaround solution that works, but is better than what is available now.